voxblox/include/voxblox/utils/color_maps.h

```diff
@@ -91,6 +91,72 @@ class IronbowColorMap : public ColorMap {
   float increment_;
 };
 
+class IdColorMap {
+ public:
+  IdColorMap() {}
+  virtual ~IdColorMap() {}
+
+  virtual Color colorLookup(size_t value) const = 0;
+};
+/**
+ * Map unique IDs from [0, Inf[ to unique colors that have constant (irrational)
+ * spacing and high spread on the color wheel.
+ * An important advantage of this color map is that the colors are independent
+ * of the max ID value, e.g. previous colors don't change when adding new IDs.
+ */
+class IrrationalIdColorMap : IdColorMap {
+ public:
+  IrrationalIdColorMap() : irrational_base_(3.f * M_PI) {}
+
+  void setIrrationalBase(float value) { irrational_base_ = value; }
+
+  virtual Color colorLookup(size_t value) const {
+    const float normalized_color = std::fmod(value / irrational_base_, 1.f);
+    return rainbowColorMap(normalized_color);
+  }
+
+ private:
+  float irrational_base_;
+};
+/**
+ * Map unique IDs from [0, Inf[ to unique colors
+ * with piecewise constant spacing and higher spread on the color wheel.
+ * An important advantage of this color map is that the colors are independent
+ * of the max ID value, e.g. previous colors don't change when adding new IDs.
+ */
+class ExponentialOffsetColorMap : IdColorMap {
+ public:
+  ExponentialOffsetColorMap() : items_per_revolution_(10u) {}
+
+  void setItemsPerRevolution(uint value) { items_per_revolution_ = value; }
+
+  virtual Color colorLookup(size_t value) const {
```

Comment:
Here and above, I personally make pass-by-value arguments const. It's saved me a few times from making assignment-instead-of-comparison mistakes.

Reply:
Good call, thanks.